### PR TITLE
Hosted docs

### PR DESCRIPTION
**Pull request by dima618**
Added modified compodoc configuration to include swagger. Also added compodoc to github pages to host documentation

### CODE REVIEW

**Comment by smalhotra3599** [comment]
Lets wait to merge this until we discuss the CI/CD aspect of the auto documentation. While the documentation is great, I don't think we need to commit all 135 files to master. For anything auto generated its typically best practice to keep in the git ignore like node_modules for packages and dist for our build directory. Our CI/CD pipeline can generate the docs at run time without having to commit their files (at least I think)

**Comment by smalhotra3599** [comment]
[CI/CD documentation for backend with github wiki - integrate with swagger/compodoc/postman/other](https://trello.com/c/LvdMcNUx/144-ci-cd-documentation-for-backend-with-github-wiki-integrate-with-swagger-compodoc-postman-other)

**Comment by dima618** [comment]
> Lets wait to merge this until we discuss the CI/CD aspect of the auto documentation. While the documentation is great, I don't think we need to commit all 135 files to master. For anything auto generated its typically best practice to keep in the git ignore like node_modules for packages and dist for our build directory. Our CI/CD pipeline can generate the docs at run time without having to commit their files (at least I think)

Those files need to be committed to the repo so that github pages can host the docs.

**Comment by smalhotra3599** [comment]
> > Lets wait to merge this until we discuss the CI/CD aspect of the auto documentation. While the documentation is great, I don't think we need to commit all 135 files to master. For anything auto generated its typically best practice to keep in the git ignore like node_modules for packages and dist for our build directory. Our CI/CD pipeline can generate the docs at run time without having to commit their files (at least I think)
> 
> Those files need to be committed to the repo so that github pages can host the docs.

Not necessarily (I think), there's a cmd command you can run to publish to github pages. In our CI/CD pipeline we can generate the documentation and use that command to push up and publish every time something gets pushed into master. I attached this link to the trello card as well but I'll also post it here: https://medium.com/front-end-weekly/ci-cd-with-github-actions-to-deploy-on-github-pages-73e225f8f131. This specific example is for building an application and publishing the dist folder after a build but the same logic applies.

**Comment by dima618** [comment]
I've added the github actions to deploy whenever a pull request is made to master.

File: .github/workflows/peer-rec-CI.yml
Comment (smalhotra3599) [inline .github/workflows/peer-rec-CI.yml]: this is great! Lets have this run on every push to master. We don't need the hosted-docs branch necessarily and we don't want to update the live docs until we get an approval. PRs going forward may have comment standards to help with auto documentation generation but you guys can decide on that as a team

File: .github/workflows/peer-rec-CI.yml
Comment (smalhotra3599) [inline .github/workflows/peer-rec-CI.yml]: more of a question - are these steps running in sequence or in parallel? Either is fine for now but eventually the building/deployment and documenting scripts should be ran concurrently as part of our CD script on a push to master. Our CI script would run on every pull request and would also run concurrently. The jobs/checks for a pull request include developer approval, passing tests, and a successful build.

**Review by smalhotra3599 — CHANGES_REQUESTED** [review]

**Comment by smalhotra3599** [comment]
[CI/CD documentation for backend with github wiki - integrate with swagger/compodoc/postman/other](https://trello.com/c/LvdMcNUx/144-ci-cd-documentation-for-backend-with-github-wiki-integrate-with-swagger-compodoc-postman-other)

File: .github/workflows/peer-rec-CI.yml
Comment (dima618) [reply .github/workflows/peer-rec-CI.yml]: They're running in sequence. I think we want to generate new docs only if tests/builds pass.

**Comment by smalhotra3599** [comment]
do swagger-docs and swagger-assets need to be merged into master or are they autogenerated files? should they be in our .gitignore like docs is

**Review by matt-weiner — DISMISSED** [review]

**Review by smalhotra3599 — APPROVED** [review]

**Pull request merged**
